pytensor/link/numba/linker.py

```diff
@@ -1,26 +1,15 @@
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING
 
-import numpy as np
-
-import pytensor
 from pytensor.link.basic import JITLinker
 
 
 if TYPE_CHECKING:
-    from pytensor.graph.basic import Variable
+    pass
 
 
 class NumbaLinker(JITLinker):
     """A `Linker` that JIT-compiles NumPy-based operations using Numba."""
 
-    def output_filter(self, var: "Variable", out: Any) -> Any:
-        if not isinstance(var, np.ndarray) and isinstance(
-            var.type, pytensor.tensor.TensorType
-        ):
-            return var.type.filter(out, allow_downcast=True)
-
-        return out
-
```

Comment:
This was actually wrong, It probably was meant to be `if not isinstance(out, np.ndarray)`, so it always triggered `var.type.fliter` which was quite slow.

Anyway, if numba is returning a non-array where we expected an array it means something is wrong in our dispatch, and we should fix it there.